packages/cozy-dataproxy-lib/src/search/SearchEngine.ts

```diff
@@ -54,48 +54,42 @@ export class SearchEngine {
     this.searchIndexes = {} as SearchIndexes
 
     this.isLocalSearch = !!getPouchLink(this.client)
-    log.info('Use local data on trusted device : ', this.isLocalSearch)
+    log.info('Use local data on trusted device: ', this.isLocalSearch)
 
     this.debouncedReplication = (): void => {
       const pouchLink = getPouchLink(client)
       if (pouchLink) {
         pouchLink.startReplicationWithDebounce()
       }
     }
-    void this.indexDocuments()
+
+    if (this.client.isLogged) {
+      this.afterLogin()
+    } else {
+      this.client.on('login', () => {
+        this.afterLogin()
+      })
+    }
```

Comment:
In the future we may want to call `indexDocuments()`/`afterLogin()` from outside of the constructor. Calling async code from a constructor is a bad practice.

Reply:
How would you do to avoid this async call ? Change the API, do have somehting like this?
```
const searchEngine = new SearchEngine()
await searchEngine.init()
```